x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/hooks/use_definition.ts

```diff
@@ -127,17 +134,40 @@ export const useDefinition = (definition: ReadStreamDefinition, refreshDefinitio
 };
 
 const createId = htmlIdGenerator();
-const createProcessorsList = (processors: ProcessingDefinition[]): ProcessorDefinition[] =>
-  processors.map(createProcessorWithId);
-
-const createProcessorWithId = (processor: ProcessingDefinition): ProcessorDefinition => ({
-  ...processor,
+const createProcessorsList = (
+  processors: ProcessorDefinition[]
+): EnrichmentUIProcessorDefinition[] => processors.map(createProcessorWithId);
+
+const createProcessorWithId = (
+  processor: ProcessorDefinition
+): EnrichmentUIProcessorDefinition => ({
+  condition: getProcessorConfig(processor).if,
+  config: {
+    ...('grok' in processor
+      ? { grok: omit(processor.grok, 'if') }
+      : { dissect: omit(processor.dissect, 'if') }),
+  },
   id: createId(),
 });
 
-const removeIdFromProcessor = (processor: ProcessorDefinition): ProcessingDefinition => {
-  const { id, ...rest } = processor;
-  return rest;
+const convertUiDefinitionIntoApiDefinition = (
+  processor: EnrichmentUIProcessorDefinition
+): ProcessorDefinition => {
+  const { id: _id, config, condition } = processor;
+  if ('grok' in config) {
+    return {
+      grok: {
+        ...config.grok,
+        if: condition,
+      },
+    };
+  }
+  return {
+    dissect: {
+      ...config.dissect,
+      if: condition,
+    },
+  };
```

Comment:
All of this could have been avoided by updating the form state shape to be equivalent to the API definition, which would have been perfectly fine.

Reply:
I tried that actually, but it greatly increases the amount of changes